sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RMResources.java

```diff
@@ -29,18 +29,18 @@
     public static final String ResourceTokenNotFound = "Resource token not found.";
     public static final String Unauthorized = "Unable to authenticate the request. The request requires valid user authentication.";
     public static final String Forbidden = "Unable to proceed with the request. Please check the authorization claims to ensure the required permissions to process the request.";
-    public static final String NotFound = "Entity with the specified id does not exist in the system.";
+    public static final String NotFound = "Entity with the specified id does not exist in the system. More info: https://aka.ms/cosmosdb-tsg-not-found";
     public static final String BadRequest = "One of the input values is invalid.";
     public static final String MethodNotAllowed = "The requested verb is not supported.";
     public static final String EntityAlreadyExists = "Entity with the specified id already exists in the system.";
     public static final String PreconditionFailed = "Operation cannot be performed because one of the specified precondition is not met.";
     public static final String RequestEntityTooLarge = "The size of the response exceeded the maximum allowed size, limit the  response size by specifying smaller value for '%s' header.";
     public static final String Locked = "";
     public static final String RetryWith = "Retry the request.";
-    public static final String TooManyRequests = "The request rate is too large. Please retry after sometime.";
+    public static final String TooManyRequests = "The request rate is too large. More Request Units may be needed, so no changes were made. Please retry after sometime. Learn more: http://aka.ms/cosmosdb-error-429";
     public static final String UnexpectedResourceType = "ResourceType %s is unexpected.";
     public static final String InvalidHeaderValue = "Value '%s' specified for the header '%s' is invalid.";
-    public static final String RequestTimeout = "Request timed out.";
+    public static final String RequestTimeout = "Request timed out. More info: https://aka.ms/cosmosdb-tsg-request-timeout";
```

Comment:
Please create a new link and page java request timeouts. This link should only be returned by the service request timeouts.

Reply:
Not sure if I understand this correctly, please help here. 
On reading this TSG : https://docs.microsoft.com/en-us/azure/cosmos-db/troubleshoot-dot-net-sdk-request-timeout, it seems that this link should not only be returned from service, as requestTimeout happens on client side as well, and we throw error using this message. The TSG clearly describes the `requestTimeout` API on the client side.

Reply:
1. This link should be returned by the service: https://docs.microsoft.com/en-us/azure/cosmos-db/troubleshoot-request-timeout
2. This link should be returned by the .NET SDK: https://docs.microsoft.com/en-us/azure/cosmos-db/troubleshoot-dot-net-sdk-request-timeout

So Java SDK should have a new page just for it's client side generated request timeouts.